packages/scripts/src/upgrade-tests/voting2/2_Verify.ts

```diff
@@ -0,0 +1,60 @@
+// This script verify that the upgrade was executed correctly.
+// It can be run against a mainnet fork by spinning a node in a separate terminal with:
+// HARDHAT_CHAIN_ID=1 yarn hardhat node --fork https://mainnet.infura.io/v3/<YOUR-INFURA-KEY> --port 9545 --no-deploy
+// and then running this script with:
+// VOTING_V2_ADDRESS=<VOTING_V2_ADDRESS> \
+// yarn hardhat run ./src/upgrade-tests/voting2/2_Verify.ts --network localhost
+
+const hre = require("hardhat");
+const assert = require("assert").strict;
+
+import { Finder, Governor, Voting, VotingV2__factory } from "@uma/contracts-node/typechain/core/ethers";
+import { getContractInstance } from "../../utils/contracts";
+const { interfaceName } = require("@uma/common");
+
+const { getContractFactory } = hre.ethers;
+
+async function main() {
+  const votingV2Address = process.env["VOTING_V2_ADDRESS"];
+
+  if (!votingV2Address) throw new Error("VOTING_V2_ADDRESS not set");
+
+  const finder = await getContractInstance<Finder>("Finder");
+  const governor = await getContractInstance<Governor>("Governor");
+  const oldVoting = await getContractInstance<Voting>("Voting");
+
+  const votingV2Factory: VotingV2__factory = await getContractFactory("VotingV2");
+  const votingV2 = await votingV2Factory.attach(votingV2Address);
+
+  console.log(" 1. Validating finder registration of new voting contract addresses...");
+  assert.equal(
+    (await finder.getImplementationAddress(hre.ethers.utils.formatBytes32String(interfaceName.Oracle))).toLowerCase(),
+    votingV2Address.toLowerCase()
+  );
+  console.log("✅ Voting registered interfaces match!");
+
+  console.log(" 2. Validating deployed contracts are owned by governor...");
+  assert.equal((await votingV2.owner()).toLowerCase(), governor.address.toLowerCase());
+  console.log("✅ New Voting correctly transferred ownership!");
+
+  console.log(" 3. Old voting is validated and migrated to the correct address.");
+  assert.equal((await oldVoting.migratedAddress()).toLowerCase(), votingV2.address.toLowerCase());
+  console.log("✅ Voting has been successfully migrated!");
+
+  console.log(" 4. Validating old voting contract and finder is owned by governor...");
+  assert.equal(await oldVoting.owner(), governor.address);
+  assert.equal(await finder.owner(), governor.address);
+  console.log("✅ Old Voting & finder correctly transferred ownership back to governor!");
+
+  console.log("Verified!");
```

Comment:
some other things we might want to verify (perhaps in a `3_livecycle_verify.ts` or something(:
1) simulate we can _stake_ _unstake_ in the new voting contract
2) simulate we can _commit/reveal a vote_ in the new voting contract
3) simulate we can re-upgrade the system *(note that this one might not need you to do any code work; it might just be running the `1_Propose.ts` script again with a newly deployed voting contract address; i.e re-running the upgrade process to check we can go from `Voting.sol->VotingV2.sol->VotingV2a.sol` to a new version of the voting contract. if we can upgrade away from this version then in all cases we are safe from a breakage.

Reply:
note that these probs should _not_ be added to this PR. you can consider adding a follow-on PR that acts to "verify" the correctness of the migration. i.e the DVM _works_ in a simulated/forked enviro meant post migration.

Reply:
+1 definitely something we should have that should sit outside of the typical verification script.